11-node-frameworks/Egor_Emelyanov/index.js

```diff
@@ -0,0 +1,72 @@
+const Koa = require('koa');
+const router = require('koa-router');
+const serve = require('koa-static');
+const session = require('koa-generic-session');
+const bodyParser = require('koa-bodyparser');
+const joi = require('joi');
+
+const app = new Koa();
+const _ = router();
+app.keys = ['kek', 'lol'];
+
+let dataBase = [];
+
+const schema = joi.object({
+  id: joi.number().required(),
+  name: joi.string().required(),
+  description: joi.string().min(3),
+});
+
+app.use(serve('assets/text'));
+app.use(serve('assets/alohadance'));
+app.use(session());
+app.use(bodyParser());
+
+function increaseCounter() {
+  const { session } = this;
+  session.count = session.count || 0;
+  session.count += 1;
+}
+
+function getCounter() {
+  this.body = this.session.count;
+}
+
+app.use(async (ctx, next) => {
+  ctx.body = 'Hello world!';
+  await next();
+});
+
+app.use(async (ctx, next) => {
+  if (ctx.path !== '/favicon.ico' && ctx.method !== 'POST') {
```

Comment:
This looks like a hack, do not you think so? 

You need to count every request without making addition calls for each new endpoints in code.  So it should work by default for every new endpoint in your code. Tip: use middleware for counting, pay attention to how body-parser works